includes/class-wc-payments-utils.php

```diff
@@ -1251,74 +1251,31 @@ public static function enqueue_style( $handle, $path = '', $deps = [], $version
 	 *
-	 * @return array
-	 */
-	public static function get_language_data( $language ) {
-		require_once ABSPATH . 'wp-admin/includes/translation-install.php';
-
-		$translations = wp_get_available_translations();
-
-		if ( isset( $translations[ $language ] ) ) {
-			return [
-				'code'         => self::convert_to_server_locale( $language ),
-				'english_name' => $translations[ $language ]['english_name'] ?? $language,
-				'native_name'  => $translations[ $language ]['native_name'] ?? $language,
-			];
-		}
-
-		return [
-			'code'         => 'en_US',
-			'english_name' => 'English (United States)',
-			'native_name'  => 'English (United States)',
-		];
-	}
-
-	/**
-	 * Converts a locale to the server supported languages.
-	 *
-	 * @param string $locale The locale to convert.
-	 *
-	 * @return string Closest locale supported ('en' if NONE)
+	 * @param string $wp_locale a WordPress locale code to be converted e.g. "en_US", "pt_BR".
+	 * @return string language code compatible with wpcom e.g. "en", "pt-br".
 	 */
-	public static function convert_to_server_locale( string $locale ): string {
-		$supported = [
-			'ar',     // Arabic.
-			'de',     // German (Germany).
-			'es',     // Spanish (Spain).
-			'fr',     // French (France).
-			'he',     // Hebrew (Israel).
-			'id',     // Indonesian (Indonesia).
-			'it',     // Italian (Italy).
-			'ja',     // Japanese.
-			'ko',     // Korean.
-			'nl',     // Dutch (Netherlands).
-			'pt-br',  // Portuguese (Brazil).
-			'ru',     // Russian (Russia).
-			'sv',     // Swedish (Sweden).
-			'tr',     // Turkish (Turkey).
-			'zh-cn',  // Simplified, Singapore).
-			'zh-tw',  // Chinese Traditional (Taiwan).
-		];
-
-		// Replace '-' with '_' (used in WordPress).
-		$locale = str_replace( '_', '-', $locale );
-
-		if ( in_array( $locale, $supported, true ) ) {
-			return $locale;
-		}
-
-		// Remove the country code and try with that.
-		$base_locale = substr( $locale, 0, 2 );
-		if ( in_array( $base_locale, $supported, true ) ) {
-			return $base_locale;
+	public static function convert_to_server_locale( string $wp_locale ): string {
+		$wp_locale_lowercase                    = strtolower( $wp_locale );
+		$region_specific_wpcom_language_codes   = [ 'pt_br', 'pt-br', 'zh_tw', 'zh-tw', 'zh_cn', 'zh-cn' ];
+		$is_region_specific_wpcom_language_code = in_array( $wp_locale_lowercase, $region_specific_wpcom_language_codes, true );
+
+		$language_code = $is_region_specific_wpcom_language_code ?
+			// If it is a region-specific language code, we replace the underscore with a dash, e.g. 'pt_br' => 'pt-br'.
+			str_replace( '_', '-', $wp_locale_lowercase ) :
+			// Otherwise, we remove the country code and return only the language code, e.g. 'nl_NL' => 'nl'.
+			preg_replace( '/([-_].*)$/i', '', $wp_locale_lowercase );
+
+		if ( empty( $language_code ) ) {
+			// If the language code is empty, we return 'en' as a fallback.
+			return 'en';
```

Comment:
There's a mix of early-return and "return $language_code" style here. I recommend rewriting as early-return while here, then each case will be a single if, no nesting.

Hmm .. if this is verbatim from Jetpack code … maybe should leave as is? If so make that clear in the comments.